README.md

```diff
@@ -179,6 +179,106 @@ or if you're using Jinja:
 | Web Experience Toolkit (WET) | v4.0.56.4 |
 | Canada.ca (GCWeb)            | v12.5.0   |
 
+
+### Sign in using Microsoft
+
+By adding a few environment variables, authentication using Microsoft's
+identity platform is automatically configured via the [Authlib](https://docs.authlib.org/en/latest/)
+library.  When enabled by setting the `PHAC_ASPC_OAUTH_PROVIDER` variable to
+"microsoft", two new routes are added:
+
+  - phac_aspc_helper_login  (`/phac_aspc_helper_login`)
+  - phac_aspc_authorize (`/phac_aspc_helper_authorize`)
+
+You must add the authorize URL `/phac_aspc_helper_authorize` to the list of 
+redirect URLs in the App Registration in Azure.
+
+Sending the browser to `/phac_aspc_helper_login` will trigger the authentication
+flow.  After successful authentication, a new user is created using the
+Azure uuid as the username.  To override this behaviour, add the following to
+your `settings.py`:
+
+```python
+
+PHAC_ASPC_OAUTH_USE_BACKEND = "custom.authentication.backend"
+
+from phac_aspc.django.settings import *
+```
+
+Here is an example backend that sets the user's name to the value provided
+by the identity service.
+
+```python
+from typing import Any
+
+from django.contrib.auth import get_user_model
+from django.contrib.auth.backends import BaseBackend
+from django.contrib.auth.base_user import AbstractBaseUser
+from django.http.request import HttpRequest
+
+
+class OAuthBackend(BaseBackend):
+    def _get(self, user_info, value, default=""):
+        return user_info[value] if value in user_info else default
+
+    def _should_update(self, user_info, value, current):
+        v = self._get(user_info, value)
+        return v != "" and v != current
+
+    def _sync_user(self, user, user_info, force=False):
+        if (
+            force
+            or self._should_update(user_info, "email", user.email)
+            or self._should_update(user_info, "name", user.first_name)
+        ):
+            user.email = self._get(user_info, "email", user.email)
+            user.first_name = self._get(user_info, "name", user.first_name)
+            user.save()
+
+    def authenticate(
+        self,
+        request: HttpRequest,
+        user_info: dict | None = None,
+        **kwargs: Any,
+    ) -> AbstractBaseUser | None:
+        if user_info is not None:
+            user_model = get_user_model()
+            try:
+                user = user_model.objects.get(username=user_info["oid"])
+                self._sync_user(user, user_info)
+            except user_model.DoesNotExist:
+                user = user_model(username=user_info["oid"])
+                self._sync_user(user, user_info, True)
+            return user
+        return None
+
+    def get_user(self, user_id):
+        user_model = get_user_model()
+        try:
+            return user_model.objects.get(pk=user_id)
+        except user_model.DoesNotExist:
+            return None
+
+```
+
+#### Environment Variables
```

Comment:
Also add the settings that can be added in settings.py - the redirect and the backend class